File: core/src/main/java/com/hsicen/core/linklist/ListNode.java
package com.hsicen.core.linklist;

/**
 * <p>作者：Night  2019/2/1 16:19
 * <p>邮箱：codinghuang@163.com
 * <p>作用：
 * <p>描述：LeetCode
 * <p>
 * 单链表定义
 */
public class ListNode {
    /*** 结点值*/
    public int val;
    /*** 下一个结点*/
    public ListNode next = null;

    public ListNode(int data) {
        this.val = data;
    }
}

File: code/src/main/java/com/hsicen/code/linklist/ListNode.kt
package com.hsicen.code.linklist

/**
 * @author: hsc
 * @date: 2025/7/3 18:29
 * @email: codinghuang@163.com
 * @description: ListNode结点定义
 */
data class ListNode(
    var `val`: Int,
    var next: ListNode? = null
) {

    fun printNode(head: ListNode?) {
        var cur = head
        while (cur != null) {
            print("${cur.`val`} ")
            cur = cur.next
        }
        println()
    }

    fun printNode(msg: String, head: ListNode?) {
        var cur = head
        print(msg)
        while (cur != null) {
            print("${cur.`val`} ")
            cur = cur.next
        }
        println()
    }
}